### ISSUE TITLE
Improve notification bar accessibility

### ISSUE THREAD

**Issue by craigcook** [mozilla/protocol#587]
The notification bar includes an optional "close" button (for when it's used as a dismissable dialog) but our examples don't have a readable text label in that button. A screen reader might skip over the button entirely, or announce it without stating what it does since its purpose isn't clear without a label. The button should have an accessible label like "Close notification" that can be visually hidden.

We're also showing static examples with a button that wasn't inserted via JavaScript, so the button in those examples does nothing. Of course they're just static examples, but that isn't made clear by the documentation so someone could easily copy-paste the markup and plonk a static bar on their site with a non-functional, inaccessible button that shouldn't be there. We can probably solve this just by adding more usage guidelines in the docs.

We could also add an aria role to the bar: `role="dialog"` for non-critical messages and `role="alertdialog"` for critical errors.

**Comment by alexgibson** [comment]
See also https://github.com/mozilla/protocol/issues/459

**Comment by craigcook** [comment]
Added comments to https://github.com/mozilla/protocol/issues/459 and we can close this as a dupe.
